tools/packgen/src/PackGen.cpp

```diff
@@ -277,12 +285,20 @@ void PackGen::ParseManifestApis(const YAML::Node node, packInfo& pack) {
 }
 
 bool PackGen::ParseManifestComponents(const YAML::Node node, packInfo& pack) {
+  uint32_t nodes = CountNodes(node, "components");
+  if (nodes >= 1) {
+    m_noComponents = false;
+  }
+  if (nodes > 1) {
+    cerr << "packgen warning: multiple 'components' sections were defined, data can be lost" << endl;
+  }
+
   YAML::Node components = node["components"];
   for (const auto& item : components) {
     const string& name = item["name"].as<string>();
 
     if (m_components.find(name) != m_components.end()) {
-      cerr << "packgen warning: component '" << name << "' has been defined multiple times" << endl;
+      cerr << "packgen warning: component '" << name << "' was defined multiple times" << endl;
```

Comment:
I would use present tense "is defined multiple times" as this is what is present in the yaml file.